packages/firebase_ml_vision/android/src/main/java/io/flutter/plugins/firebasemlvision/FirebaseMlVisionPlugin.java

```diff
@@ -1,26 +1,123 @@
 package io.flutter.plugins.firebasemlvision;
 
+import android.graphics.Point;
+import android.graphics.Rect;
+import android.net.Uri;
+import android.support.annotation.NonNull;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.firebase.ml.vision.FirebaseVision;
+import com.google.firebase.ml.vision.common.FirebaseVisionImage;
+import com.google.firebase.ml.vision.text.FirebaseVisionText;
+import com.google.firebase.ml.vision.text.FirebaseVisionTextDetector;
 import io.flutter.plugin.common.MethodCall;
 import io.flutter.plugin.common.MethodChannel;
 import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
 import io.flutter.plugin.common.MethodChannel.Result;
 import io.flutter.plugin.common.PluginRegistry.Registrar;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
 /** FirebaseMlVisionPlugin */
 public class FirebaseMlVisionPlugin implements MethodCallHandler {
+  private Registrar registrar;
+
+  private FirebaseMlVisionPlugin(Registrar registrar) {
+    this.registrar = registrar;
+  }
+
   /** Plugin registration. */
   public static void registerWith(Registrar registrar) {
     final MethodChannel channel =
         new MethodChannel(registrar.messenger(), "plugins.flutter.io/firebase_ml_vision");
-    channel.setMethodCallHandler(new FirebaseMlVisionPlugin());
+    channel.setMethodCallHandler(new FirebaseMlVisionPlugin(registrar));
   }
 
   @Override
   public void onMethodCall(MethodCall call, Result result) {
-    if (call.method.equals("getPlatformVersion")) {
-      result.success("Android " + android.os.Build.VERSION.RELEASE);
+    if (call.method.equals("TextDetector#detectInImage")) {
+      handleTextDetectionResult(call, result);
     } else {
       result.notImplemented();
     }
   }
+
+  private void handleTextDetectionResult(MethodCall call, final Result result) {
+    File file = new File((String) call.arguments);
+
+    FirebaseVisionImage image;
+    try {
+      image = FirebaseVisionImage.fromFilePath(registrar.context(), Uri.fromFile(file));
+    } catch (IOException exception) {
+      exception.printStackTrace();
```

Comment:
Generally we avoid printing in the plugins. Just pass the exception up and let it be handled by the caller.